libnmstate/dns.py

```diff
@@ -147,7 +147,7 @@ def _find_ifaces_for_name_servers(self, ifaces, route_state):
 
         return ipv4_iface, ipv6_iface
 
-    def _find_ifaces_with_static_gateways(self, route_state):
+    def _find_ifaces_with_static_gateways(self, ifaces, route_state):
```

Comment:
At the time function `_find_ifaces_with_static_gateways()` been invoked, we don't know the ignored interface yet.

The metadata generate happens at `NetState.__init__()`, but `_apply_ifaces_state()` who mark ignored interfaces is after that.

Reply:
That is correct, this is why I added this on the `NetState.__init__()`.

```
        if not gen_conf_mode:
            self._mark_ignored_kernel_ifaces(plugins)
```

Please, check how we are marking the interface as ignored earlier on this patch.